databricks/koalas/indexes.py

```diff
@@ -1962,6 +1962,56 @@ def holds_integer(self):
         """
         return isinstance(self.spark.data_type, IntegralType)
 
+    def intersection(self, other):
+        """
+        Form the intersection of two Index objects.
+
+        This returns a new Index with elements common to the index and `other`.
+
+        Parameters
+        ----------
+        other : Index or array-like
+
+        Returns
+        -------
+        intersection : Index
+
+        Examples
+        --------
+        >>> idx1 = ks.Index([1, 2, 3, 4])
+        >>> idx2 = ks.Index([3, 4, 5, 6])
+        >>> idx1.intersection(idx2).sort_values()
+        Int64Index([3, 4], dtype='int64')
+        """
+        if isinstance(other, Index):
+            if type(self) != type(other):
+                # If self Index and MultiIndex coexist on both sides,
+                # it always returns an empty MultiIndex.
+                return MultiIndex.from_tuples([(0, 0)]).take([])
+            spark_frame_other = other._internal.spark_frame.drop(NATURAL_ORDER_COLUMN_NAME)
+        elif isinstance(other, Series):
+            spark_frame_other = other._kdf.to_spark()
```

Comment:
I think you should `other.to_frame().to_spark()` to keep the changes in Spark column from the Series. Otherwise, the changes that have been made in Series will be ignored.

Reply:
Maybe don't we update the anchor when we change the Series ?